File: csharp/resistor-color/ResistorColor.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ResistorColor
{
    private static readonly List<string> _colors = new()
    {
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue", 
        "violet",
        "grey",
        "white"
    };
    
    public static int ColorCode(string color) =>
        _colors.IndexOf(color);

    public static string[] Colors() =>
        _colors.ToArray();
}

File: c/resistor-color/Iteration1/resistor_color.c

#include <stddef.h>

#include "resistor_color.h"

static resistor_band_t _colors[] = {
    BLACK,
    BROWN,
    RED,
    ORANGE,
    YELLOW,
    GREEN,
    BLUE,
    VIOLET,
    GREY,
    WHITE
};

unsigned int color_code(resistor_band_t resistorBand)
{
    int result = 0;

    for (resistor_band_t* color = _colors; color != NULL; color++)
    {
        if (*color == resistorBand) break;
        result++;
    }

    return result;
}

resistor_band_t* colors()
{
    return _colors;
}